aten/src/ATen/CPUGeneral.cpp

```diff
@@ -1,16 +1,149 @@
+#include <ATen/Config.h>
 #include <ATen/CPUGeneral.h>
+#include <c10/util/Exception.h>
 #include <atomic>
 #include <memory>
 #include <thread>
+#include <cstdint>
+#include <mutex>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#if AT_MKL_ENABLED()
+#include <mkl.h>
+#endif
+
 
 namespace at {
-// Lock free atomic type
-std::atomic<int> num_threads(-1);
 
-void set_num_threads(int num_threads_) {
-  if (num_threads_ >= 0)
-    num_threads.store(num_threads_);
+// Global value
+// This is probably lock-free, but is_lock_free is determined at runtime so we
+// can't static_assert. See notes at https://en.cppreference.com/w/cpp/atomic/atomic_is_lock_free
+std::atomic<int> num_threads_global(-1);
+
+// Global lock
+std::mutex num_threads_global_mutex;
```

Comment:
unused?

Reply:
yes unused. I should clean up this. But the more important thing is to figure out why the JIT test fails...